src/Compilers/CSharp/Portable/Symbols/NamespaceOrTypeSymbol.cs

```diff
@@ -334,6 +343,8 @@ public virtual ImmutableArray<NamedTypeSymbol> GetTypeMembers(string name, int a
 Done:
             if (isTopLevel
                 && (emittedTypeName.ForcedArity == -1 || emittedTypeName.ForcedArity == emittedTypeName.InferredArity)
+                // Quick check so we don't have to realize the expensive `UnmangledTypeName` in the common case.
+                && emittedTypeName.TypeNameMemory.Span is ['<', ..]
                 && GeneratedNameParser.TryParseFileTypeName(
                     emittedTypeName.UnmangledTypeName,
                     out string? displayFileName,
```

Comment:
this is the rare case @RikkiGibson , so rather than virally have to update GeneratedNameParser.TryParseFileTypeName, i just make it so that we rarely need to go into there, and we pay no cost when we don't.

Reply:
fwiw, I think this method has just two usages to update, and having it output ReadOnlyMemory<char> which one or the other caller needs to convert to string in certain cases is fine.

Reply:
I wasn't sure if the regex stuff you did in that type works with ROM/Span (esp. on netstandard).  But i can look to see :)